File: spring_security/eg05_sb3_oauth2_resource_server_jwt_symmetric_key/src/main/java/com/sc/sb/auth/repository/ScUserRepository.java
package com.sc.sb.auth.repository;

import com.sc.sb.auth.entity.ScUser;
import org.springframework.data.repository.CrudRepository;

import java.util.Optional;

public interface ScUserRepository extends CrudRepository<ScUser, Long> {
    Optional<ScUser> findByUserNameIgnoreCase(String userName);
}

File: spring_security/eg06_kotlin_sb3_oauth2_resource_server_jwt_symmetric_key/src/main/kotlin/com/sc/sb/auth/repository/ScUserRepository.kt
package com.sc.sb.auth.repository

import com.sc.sb.auth.entity.ScUser
import org.springframework.data.repository.CrudRepository
import java.util.Optional

interface ScUserRepository: CrudRepository<ScUser, Long> {
    fun findByUserNameIgnoreCase(userName: String): Optional<ScUser>
}